.github/workflows/test-e2e-ios.yaml

```diff
@@ -29,18 +29,60 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  e2e-ios:
+  build-release:
     runs-on: macos-15
+    outputs:
+      artifact-id: ${{ steps.rnef-remote-build-ios.outputs.artifact-id }}
+      artifact-url: ${{ steps.rnef-remote-build-ios.outputs.artifact-url }}
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup
+        uses: ./.github/actions/setup
+
+      - name: Run generate:legal from bare-example
+        run: yarn workspace react-native-legal-bare-example generate:legal
+        shell: bash
+
+      - name: Build iOS
+        id: rnef-remote-build-ios
+        uses: callstackincubator/ios@v1
+        with:
+          destination: 'simulator'
+          scheme: 'ReactNativeLegalBareExample'
+          configuration: 'Release'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          working-directory: ./examples/bare-example
+          comment-bot: false
+
+  run-e2e-tests:
+    runs-on: macos-15
+    needs: build-release
     strategy:
       fail-fast: false
       matrix:
-        simulator: ['iPhone 16 Pro (18.5)', 'iPhone SE (3rd generation) (18.0)']
+        simulator: ['iPhone 16 Pro', 'iPhone SE (3rd generation)']
+
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Download and Unpack APK artifact
+        run: |
+          curl -L -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -o artifact.zip "https://api.github.com/repos/${{ github.repository }}/actions/artifacts/${{ needs.build-release.outputs.artifact-id }}/zip"
+          unzip artifact.zip -d downloaded-artifacts
+          tar zxf downloaded-artifacts/app.tar.gz -C downloaded-artifacts
+          ls -l downloaded-artifacts
+          APP_PATH=$(pwd)/$(find downloaded-artifacts -name "*.app" -print -quit)
+          echo "ARTIFACT_PATH_FOR_E2E=$APP_PATH" >> $GITHUB_ENV
+        shell: bash
```

Comment:
instead of doing custom logic you can use `remote-cache` command from: https://github.com/callstack/rnef/pull/263

Reply:
You can find the docs here now: https://www.rnef.dev/docs/cli#rnef-remote-cache-actions-and-options